src/ignoreFileHandler.ts

```diff
@@ -0,0 +1,46 @@
+const ignore = require('ignore');
+import { readFileSync, existsSync } from 'fs';
+import * as path from 'path';
+import { workspace } from 'vscode';
+
+interface Ignorer {
+    ignores(filePath: string): boolean;
+}
+
+const nullIgnorer: Ignorer = { ignores: () => false };
+
+/**
+ * Create a file watcher. Reloads the contents of .prettierignore
+ * on file change, create, delete.
+ */
+function fileListener(ignorePath: string) {
+    const fileWatcher = workspace.createFileSystemWatcher(ignorePath);
+
+    fileWatcher.onDidChange(reloadIgnoreFile);
+    fileWatcher.onDidCreate(reloadIgnoreFile);
+    fileWatcher.onDidDelete(reloadIgnoreFile);
+
+    let ignorer: Ignorer;
+    reloadIgnoreFile();
+
+    return {
+        fileWatcher,
+        fileIsIgnored(filePath: string) {
+            return ignorer.ignores(filePath);
+        },
+    };
+
+    function reloadIgnoreFile() {
+        ignorer = nullIgnorer;
+
+        if (workspace.rootPath) {
```

Comment:
This will change in future VSCode and multiple workspace.

Reply:
For reference:
https://github.com/Microsoft/vscode/wiki/Extension-Authoring:-Adopting-Multi-Root-Workspace-APIs

Reply:
Seems like it will take a fair amount of work to support settings at a `resource` level, or am I reading this wrong?

Reply:
I think all but `{LANGUAGE}Enable` settings would be quite easy. We read them on format and we have file's path (except for none persisted files).

 `ignorePath` will be a little trickier :-)
Options I see:
- With your pattern it will need to have an ignorer per workspace and find out to which workspace correspond a given file
- Read file the right file at each call (no caching). Certainly easier but also certainly worse performance wise.